### PR TITLE
hdf5 hdf5-mpi 1.14.0

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
```
CMake Error in src/CMakeLists.txt:
  Imported target "hdf5::hdf5-shared" includes non-existent path

    "/private/tmp/hdf5-20230101-55793-1m01qsy/hdf5-1.14.0/src/H5FDsubfiling"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.
```

**Comment by cho-m** [comment]
`hdf5` configure script is adding the build directory path which gets saved in installed files:
```console
❯ rg /private/tmp
share/hdf5/libhdf5.settings
40:                    AM_CPPFLAGS:  -I@@HOMEBREW_PREFIX@@/opt/libaec/include -I/private/tmp/hdf5-20230101-49316-1941juc/hdf5-1.14.0/src/H5FDsubfiling

bin/h5c++
90:H5BLD_CPPFLAGS=" -I@@HOMEBREW_PREFIX@@/opt/libaec/include -I/private/tmp/hdf5-20230101-49316-1941juc/hdf5-1.14.0/src/H5FDsubfiling "

bin/h5cc
92:H5BLD_CPPFLAGS=" -I@@HOMEBREW_PREFIX@@/opt/libaec/include -I/private/tmp/hdf5-20230101-49316-1941juc/hdf5-1.14.0/src/H5FDsubfiling "
```

Probably from new feature https://github.com/HDFGroup/hdf5/commit/27bb358f7ab1d23f3f8ce081c6b4f1602033e4d7

Particularly following https://github.com/HDFGroup/hdf5/blob/hdf5-1_14_0/configure.ac#L3014-L3016
```
## Always include subfiling directory so public header files are available
CPPFLAGS="$CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
AM_CPPFLAGS="$AM_CPPFLAGS -I$ac_abs_confdir/src/H5FDsubfiling"
```

**Comment by fxcoudert** [comment]
May have to be rerun after https://github.com/Homebrew/homebrew-core/pull/129772 is merged

**Comment by fxcoudert** [comment]
Will need a vtk revision bump.

**Review by SMillerDev — DISMISSED** [review]

**Review by SMillerDev — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Okay, so if not a manual merge queue. How am I supposed to merge this?

> Commit style — a62df7a935 modifies non-formula files (maintainers must merge manually)

suggests a manual merge, but I'm not sure how to do that @carlocab

File: Formula/hdf5.rb [hunk 1]
```diff
@@ -112,3 +115,28 @@ def install
     assert_equal version.to_s, shell_output("./a.out").chomp
   end
 end
+__END__
```
Comment (carlocab) [inline Formula/hdf5.rb:118]:
```suggestion
end

__END__
```

**Review by carlocab — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Great, a conflict with Octave now

**Comment by fxcoudert** [comment]
Rebased and pushed again to CI

**Review by SMillerDev — DISMISSED** [review]

**Comment by fxcoudert** [comment]
Linux: dynare build failure with

```
  CodeInterpreter.hh:1565:32: error: 'uint8_t' does not name a type
   1565 |         || type == static_cast<uint8_t>(BlockSimulationType::solveForwardComplete))
        |                                ^~~~~~~
  CodeInterpreter.hh:1565:32: note: 'uint8_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
```

macOS 13 Intel: `sratoolkit` test failure (seems intermittent, seen it in other PRs)

**Comment by carlocab** [comment]
I'll dispatch another build for `sratoolkit` after merging. Dynare has only 3 downloads on Linux in the past year, so probably not worth delaying this. I'll open a PR for fixes.

**Comment by fxcoudert** [comment]
```
Error: docker://ghcr.io/homebrew/core/cdo:2.2.0_1 already exists!
```

😭

**Comment by fxcoudert** [comment]
And now:

```
/usr/bin/env /usr/bin/skopeo inspect --raw docker://ghcr.io/homebrew/core/alpscore:2.2.0_3 --creds=brewtestbot:******
Warning: docker://ghcr.io/homebrew/core/alpscore:2.2.0_3 already exists, skipping upload!

/usr/bin/env /usr/bin/skopeo inspect --raw docker://ghcr.io/homebrew/core/csound:6.18.1_2 --creds=brewtestbot:******
Warning: docker://ghcr.io/homebrew/core/csound:6.18.1_2 already exists, skipping upload!
```

**Comment by fxcoudert** [comment]
And now it's complaining about the lack of long timeout label (which it shouldn't need at this stage). Should we add `CI-published-bottle-commits` manually?

**Review by fxcoudert — APPROVED** [review]

**Comment by carlocab** [comment]
Bottles weren't correctly uploaded for some reason. Let me see what happened...

**Comment by carlocab** [comment]
Ok, got it. Fix in progress...

**Comment by carlocab** [comment]
Almost ready to upload bottles...

**Review by carlocab — APPROVED** [review]

**Pull request merged**